src/main/java/uk/ac/ebi/eva/utils/MongoDBHelper.java

```diff
@@ -129,8 +47,7 @@ private MappingMongoConverter getMappingMongoConverter(MongoDbFactory mongoFacto
         return serverAddresses;
     }
 
-
-    private ReadPreference getMongoTemplateReadPreferences(String readPreference) {
+    public static ReadPreference getMongoTemplateReadPreferences(String readPreference) {
         switch (readPreference) {
             case "primary":
```

Comment:
In fact, the `ReadPreference` enum from the Mongo API can deal with this perfectly, it even throws an `IllegalArgumentException` if an incorrect value is provided.

I will replace this method with a call to `ReadPreference.valueOf(readPreference)`.